### PR TITLE
Fixes ai xenos targeting hiveminds and other xenos at long distances, structure/door interaction fixed

### PR DESCRIPTION
**Pull request by ma44**
## About The Pull Request
Fixes some bugs left behind by ai framework

## Why It's Good For The Game
Makes ai xenos work ever so better

## Changelog
:cl: ma44
fix: AI xenomorphs no longer try to attack immune-to-damage hiveminds
fix: AI xenos also will now actually attack windows and open airlocks
/:cl:

### CODE REVIEW

File: code/__HELPERS/ai.dm [hunk 1]
```diff
@@ -20,3 +20,15 @@
...
+/proc/cheap_get_xenos_near(atom/movable/source, distance)
+	. = list()
+	for(var/xeno in GLOB.alive_xeno_list)
+		var/mob/living/carbon/xenomorph/nearby_xeno = xeno
+		if(source.z != nearby_xeno.z)
+			continue
+		if(get_dist(source, nearby_xeno) > distance)
+			continue
+		. += nearby_xeno
+	return .
```
Comment (Rohesie) [inline code/__HELPERS/ai.dm:34]:
```suggestion
```

File: code/modules/ai/ai_behaviors/carbon/xeno/xeno.dm [hunk 2]
```diff
@@ -46,7 +51,7 @@
 
 	var/list/things_nearby = range(mob_parent, 1) //Rather than doing multiple range() checks we can just archive it here for just this deal_with_obstacle
 	for(var/obj/structure/obstacle in things_nearby)
-		if(!(obstacle.resistance_flags & XENO_DAMAGEABLE))
+		if(obstacle.resistance_flags && XENO_DAMAGEABLE)
```
Comment (Rohesie) [inline code/modules/ai/ai_behaviors/carbon/xeno/xeno.dm:54]:
```suggestion
		if(obstacle.resistance_flags & XENO_DAMAGEABLE)
```

File: code/modules/ai/ai_behaviors/carbon/xeno/xeno.dm [hunk 3]
```diff
@@ -16,9 +16,15 @@
...
-	for(var/x in GLOB.alive_xeno_list)
-		if(!xeno_parent.issamexenohive(x)) //Xenomorphs not in our hive will attack as well!
-			return_result += x
+	for(var/x in cheap_get_xenos_near(mob_parent, 8))
+		if(xeno_parent.issamexenohive(x)) //Xenomorphs not in our hive will be attacked as well!
+			continue
+		var/mob/nearby_xeno = x
+		if(nearby_xeno.stat == DEAD)
+			continue
+		if((nearby_xeno.status_flags && GODMODE) || (nearby_xeno.status_flags && GODMODE)) //No attacking the hivemind!
```
Comment (Rohesie) [inline code/modules/ai/ai_behaviors/carbon/xeno/xeno.dm:25]: Flags don't use `&&`

Comment (ma44) [reply code/modules/ai/ai_behaviors/carbon/xeno/xeno.dm]: Resolved.

File: code/modules/ai/ai_behaviors/carbon/xeno/xeno.dm [hunk 4]
```diff
@@ -16,9 +16,15 @@
...
-	for(var/x in GLOB.alive_xeno_list)
-		if(!xeno_parent.issamexenohive(x)) //Xenomorphs not in our hive will attack as well!
-			return_result += x
+	for(var/x in cheap_get_xenos_near(mob_parent, 8))
+		if(xeno_parent.issamexenohive(x)) //Xenomorphs not in our hive will be attacked as well!
+			continue
+		var/mob/nearby_xeno = x
+		if(nearby_xeno.stat == DEAD)
+			continue
+		if((nearby_xeno.status_flags & GODMODE) || (nearby_xeno.status_flags & GODMODE)) //No attacking the hivemind!
```
Comment (psykzz) [inline code/modules/ai/ai_behaviors/carbon/xeno/xeno.dm:25]:
```suggestion
		if((nearby_xeno.status_flags & INCORPOREAL) || (nearby_xeno.status_flags & INCORPOREAL)) //No attacking the hivemind!
```

**Comment by spookydonut** [comment]
@MerryweatherLost this is broken dont TM it.

**Comment by ma44** [comment]
@MerryweatherLost Should be good for testmerges again

If something somehow breaks pinging me on discord or reporting here would be great, as the xenomorphs seem to work fine when i play with them locally (can shoot, can die from them, abilities work, they die without leaving runtimes or still being referenced etc.)

File: code/modules/ai/ai_behaviors/carbon/xeno/xeno.dm (hunk 4 above)
Comment (ma44) [reply code/modules/ai/ai_behaviors/carbon/xeno/xeno.dm]: INCORPOREAL and GODMODE are now both checked

**Pull request merged**